setup/tomcat/conf/conf.d/setmem.conf

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+function calculate_heapsize_mb() {
+    local USAGE_PERCENT METASPACE_SIZE_MB METASPACE_SIZE_KB TOTAL_MEM_KB HEAP_KB HEAP_MB
+    USAGE_PERCENT=$1
+    METASPACE_SIZE_MB=$2
+    TOTAL_MEM_KB="$(awk '/MemTotal:/ { print $2 }' /proc/meminfo)"
+
+    (( METASPACE_SIZE_KB=METASPACE_SIZE_MB / 1024 ))
+    (( HEAP_KB=(TOTAL_MEM_KB - METASPACE_SIZE_KB) * USAGE_PERCENT / 100 ))
+    (( HEAP_MB=HEAP_KB / 1024 ))
+
+    echo "$HEAP_MB"
+}
+
+# Set default and maximum metaspace size
+METASPACE_SIZE_MB="384"
+METASPACE_MAX_SIZE_MB="512"
+
+CATALINA_OPTS="$CATALINA_OPTS -XX:MetaspaceSize=${METASPACE_SIZE_MB}m"
+CATALINA_OPTS="$CATALINA_OPTS -XX:MaxMetaspaceSize=${METASPACE_MAX_SIZE_MB}m"
+CATALINA_OPTS="$CATALINA_OPTS -XX:CompressedClassSpaceSize=128m"
+
+# Set maximum and minimum heap size (80% from total server memory minus metaspace size)
+HEAP_SIZE_MB="$(calculate_heapsize_mb 80 "$METASPACE_MAX_SIZE_MB")"
```

Comment:
@Simplycissimys, I think we talked about 60% of total memory for the heap. Right?

Reply:
Changed to 70%, issue comments contains this number.

Reply:
70% is ok